packages/@aws-cdk-testing/cli-integ/lib/cli/run-suite.ts

```diff
@@ -2,115 +2,126 @@
+        alias: 'f',
+        type: 'string',
+      })
+      .option('use-source', {
+        descripton: 'Use TypeScript packages from the given source repository (or "auto")',
+        type: 'string',
+        requiresArg: true,
+        deprecated: 'Use --cli-source instead',
+      })
+      .option('use-cli-release', {
+        descripton: 'Run the current tests against the CLI at the given version',
+        alias: 'u',
+        type: 'string',
+        requiresArg: true,
+        deprecated: 'Use --cli-version instead',
+      })
+      .option('auto-source', {
+        alias: 'a',
+        describe: 'Automatically find the source tree from the current working directory',
+        type: 'boolean',
+        requiresArg: false,
+        deprecated: 'Use --use-source=auto instead',
+      })
+      /////////////////////////////////////////////////////////////////////////////
+      //  Test running flags
+      .option('runInBand', {
+        descripton: 'Run all tests in one Node process',
+        alias: 'i',
+        type: 'boolean',
+      })
+      .option('test', {
+        descripton: 'Test pattern to selectively run tests',
+        alias: 't',
+        type: 'string',
+        requiresArg: true,
+      })
+      .option('test-file', {
+        describe: 'The specific test file to run',
+        alias: 'F',
+        type: 'string',
+        requiresArg: true,
+      })
+      .options('verbose', {
+        alias: 'v',
+        describe: 'Run in verbose mode',
+        type: 'boolean',
+        requiresArg: false,
+      })
+      .options('passWithNoTests', {
+        describe: 'Allow passing if the test suite is not found (default true when IS_CANARY mode, false otherwise)',
+        type: 'boolean',
+        requiresArg: false,
+      })
+      .options('maxWorkers', {
+        alias: 'w',
+        describe: 'Specifies the maximum number of workers the worker-pool will spawn for running tests. We use a sensible default for running cli integ tests.',
+        type: 'string',
+        requiresArg: true,
+      }), () => {
+    },
+    )
+    .strict()
+    .parse();
+
+  const suiteName = args.SUITENAME;
 
-  let packageSource: undefined | IPackageSourceSetup;
-  function usePackageSource(s: IPackageSourceSetup) {
-    if (packageSource) {
-      throw new Error('Cannot specify two package sources');
+  // So many ways to specify this, and with various ways to spell the same flag (o_O)
+  const cliSource = new UniqueOption<IRunnerSource<ITestCliSource>>('CLI version');
+  for (const flagAlias of ['cli-version', 'use-cli-release'] as const) {
+    if (args[flagAlias]) {
+      cliSource.set(new RunnerCliNpmSource(args[flagAlias]), `--${flagAlias}`);
     }
-    packageSource = s;
   }
-
-  if (args['use-source'] || args['auto-source']) {
-    if (args['framework-version']) {
-      throw new Error('Cannot use --framework-version with --use-source');
+  for (const flagAlias of ['cli-source', 'use-source'] as const) {
```

Comment:
I think this gives `--use-source` precedence over `--cli-source` if both are defined.

Reply:
`cliSource` is of type `UniqueOption<T>` which will throw an error if it's set more than once.